lib/puppet/face/config.rb

```diff
@@ -94,16 +96,47 @@
 
         values = Puppet.settings.values(Puppet[:environment].to_sym, options[:section].to_sym)
         report_section_and_environment(options[:section], Puppet.settings[:environment])
-        if args.length == 1
-          puts values.interpolate(args[0].to_sym)
-        else
-          args.sort.each do |setting_name|
-            puts "#{setting_name} = #{values.interpolate(setting_name.to_sym)}"
-          end
+
+        to_be_rendered = {}
+        args.sort.each do |setting_name|
+          to_be_rendered[setting_name] = values.interpolate(setting_name.to_sym)
         end
       end
-      nil
+
+      # convert symbols to strings before formatting output
+      if render_all_settings
+        to_be_rendered = stringifyhash(to_be_rendered)
```

Comment:
Why are we only stringifying the hash if we are rendering all settings? If there's a subset, we'll still have symbols in the hash

Reply:
Short answer: The arg variable is storing the settings that need to be rendered. This gets populated in 2 different places in the code for the 2 different cases.
Longer answer: 
* Case 1: we have a subset (puppet config print tags --render-as yaml)
The setting is coming in from the command line. The top of 'when_invoked' args starts off as ["tags", {:section=>:main}]. After "options = args.pop", we are left with the requested setting in string form for the args variable ["tags"]
* Case 2: we are requesting all settings either explicitly or by not specifying a setting (puppet config print --render-as yaml)
To start, the args variable contains only [{:section=>:main}] leaving args empty after "options = args.pop". args then gets populated around line 69, and this returns an array like [:confdir, :codedir...]

Reply:
Oh, I see. That makes sense. I think I saw `setting_name.to_sym` on line 80 above, and that was tripping me up.

```
    76:         # the requested section
    77:         values = Puppet.settings.values(Puppet[:environment].to_sym, options[:section].to_sym)
    78:
    79:         to_be_rendered = {}
    80:         args.sort.each do |setting_name|
 => 81:           to_be_rendered[setting_name] = values.interpolate(setting_name.to_sym)
    82:         end
    83:       end
    84:
    85:       # convert symbols to strings before formatting output
    86:       if render_all_settings

[24] pry(#<Puppet::Interface>)> setting_name.to_sym
=> :environment
[25] pry(#<Puppet::Interface>)> values.interpolate(setting_name.to_sym)
=> "production"
[26] pry(#<Puppet::Interface>)> setting_name
=> "environment"
[28] pry(#<Puppet::Interface>)> to_be_rendered
=> {"environment"=>"production"}
```

but this does corroborate what you say above.


It looks like it *should* be safe to call `stringifyhash` on the hash regardless of how it gets populated
```
[30] pry(#<Puppet::Interface>)> to_be_rendered
=> {"environment"=>"production", "server"=>"puppet"}
[31] pry(#<Puppet::Interface>)> stringifyhash(to_be_rendered)
=> {"environment"=>"production", "server"=>"puppet"}
```
I'm not saying this is the best idea, but it could potentially be nice to simplify the code